met-web/src/components/engagement/view/widgets/EventsWidget.tsx

```diff
@@ -0,0 +1,156 @@
+                <Divider sx={{ borderWidth: 1, marginTop: 0.5 }} />
+            </Grid>
+            {events.map((event: Event) => {
+                const eventItem = event.event_items[0];
+
+                return (
+                    <Grid
+                        key={event.id}
+                        container
+                        item
+                        columnSpacing={3}
+                        rowSpacing={1}
+                        xs={12}
+                        paddingTop={2}
+                        paddingBottom={4}
+                    >
+                        <Grid
+                            item
+                            container
+                            justifyContent={{ xs: 'center', md: 'flex-start' }}
+                            alignItems="flex-start"
+                            direction="row"
+                            rowSpacing={1}
+                            xs={12}
+                            md={9}
+                        >
+                            <When condition={Boolean(eventItem.description)}>
+                                <Grid
+                                    container
+                                    justifyContent={{ xs: 'center', md: 'flex-start' }}
+                                    paddingBottom={1}
+                                    item
+                                    xs={12}
+                                >
+                                    <MetBody>{eventItem.description}</MetBody>
+                                </Grid>
+                            </When>
+                            <When
+                                condition={
+                                    event.type === EVENT_TYPE.MEETUP.value || event.type === EVENT_TYPE.OPENHOUSE.value
+                                }
+                            >
+                                <Grid container justifyContent={{ xs: 'center', md: 'flex-start' }} item xs={12}>
+                                    <MetBody>Location: {eventItem.location_name}</MetBody>
+                                </Grid>
+                                <Grid container justifyContent={{ xs: 'center', md: 'flex-start' }} item xs={12}>
+                                    <MetBody>Address: {eventItem.location_address}</MetBody>
+                                </Grid>
+                            </When>
+                            <Grid item container justifyContent={{ xs: 'center', md: 'flex-start' }} xs={12}>
+                                <MetBody>Date: {formatDate(eventItem.start_date, 'MMMM DD, YYYY')}</MetBody>
+                            </Grid>
+                            <Grid container justifyContent={{ xs: 'center', md: 'flex-start' }} item xs={12}>
+                                <MetBody>
+                                    Time:{' '}
+                                    {`${formatDate(eventItem.start_date, 'h:mm a')} to ${formatDate(
```

Comment:
any vetter way of handling formatDate ? without giving the format in the template..
like MMMM DD, YYYY will be in multiple places in the whole app

Reply:
same with h:mm:a